File: gateway/test/audio-checkin-backbone.test.js
const test = require('node:test');
const assert = require('node:assert/strict');

const {
  SERVICE_CONTRACT,
} = require('../src/service-backbones/audio-checkin');

test('audio-checkin remains a disabled service backbone', () => {
  assert.equal(SERVICE_CONTRACT.lifecycle, 'backbone');
  assert.equal(SERVICE_CONTRACT.enabledByDefault, false);
  assert.equal(SERVICE_CONTRACT.customerVisible, false);
  assert.equal(SERVICE_CONTRACT.minimumPlan, 'family');
  assert.ok(SERVICE_CONTRACT.protocolCommands.length > 0);
  assert.deepEqual(SERVICE_CONTRACT.documentedProtocolVariants, [
    'MONITOR',
    'MONITOR,<verified callback>',
  ]);
  assert.equal(SERVICE_CONTRACT.protocolEvidence, 'vendor-conflict-unproven');
  assert.ok(SERVICE_CONTRACT.backendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.frontendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.acceptanceGates.length > 0);
  assert.ok(Object.isFrozen(SERVICE_CONTRACT));
});

File: apps/mobile/test/audio_checkin_backbone_test.dart
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';
import 'package:guardian/services/backbones/audio_checkin_backbone.dart';

void main() {
  test('audio-checkin stays hidden until its acceptance gates pass', () {
    expect(AudioCheckinBackbone.lifecycle, 'backbone');
    expect(AudioCheckinBackbone.enabledByDefault, isFalse);
    expect(AudioCheckinBackbone.customerVisible, isFalse);
    expect(AudioCheckinBackbone.canRenderCustomerEntryPoint, isFalse);
    expect(AudioCheckinBackbone.minimumPlan, 'family');
    expect(AudioCheckinBackbone.protocolCommands, isNotEmpty);
    expect(
      AudioCheckinBackbone.documentedProtocolVariants,
      containsAll(<String>['MONITOR', 'MONITOR,<verified callback>']),
    );
    expect(
      AudioCheckinBackbone.protocolEvidence,
      'vendor-conflict-unproven',
    );
    expect(AudioCheckinBackbone.frontendMilestones, isNotEmpty);
    expect(AudioCheckinBackbone.acceptanceGates, isNotEmpty);
  });

  test('legacy direct MONITOR controls are absent from the customer app', () {
    final commandService = File(
      'lib/services/guardian_services.dart',
    ).readAsStringSync();
    final accountPage = File('lib/screens/account_page.dart').readAsStringSync();

    expect(commandService, isNot(contains('startVoiceMonitor')));
    expect(commandService, isNot(contains("'voice_monitor'")));
    expect(accountPage, isNot(contains("Text('Listen in')")));
    expect(accountPage, isNot(contains('receive the silent call')));
  });
}
